PayloadPkg/OsLoader/LoadImage.c

```diff
@@ -93,12 +93,25 @@ GetBootImageFromRawPartition (
   if (BlockData == NULL) {
     return EFI_OUT_OF_RESOURCES;
   }
-  Status = MediaReadBlocks (
-             BootOption->HwPart,
-             LogicBlkDev.StartBlock + LbaAddr,
-             AlignedHeaderSize,
-             BlockData
-             );
+
+  if ( BootOption->DevType == OsBootDeviceMemory )
+  {
+    Status = MediaReadBlocks (
+               BootOption->HwPart,
+               LogicBlkDev.StartBlock + LbaAddr,
+               AlignedHeaderSize,
+               BlockData
+               );
+  } else {
+   Status = MediaReadBlocks (
+               BootOption->HwPart,
+               LbaAddr,

```

Comment:
Simplify the code by using a new varaible for LbaAddr to call for MediaReadBlocks so that single function call can be used instead.